guides/plugins/plugins/checkout/cart/add-cart-validator.md

```diff
@@ -0,0 +1,214 @@
+This is just an example and will always result in an error, since the data requested doesn't exist by default, until you add them.
+
+A validator should be placed in the proper domain.
+That means, that an Address validator should be in a directory `<plugin root>/src/Core/Checkout/Cart/Address`.
+Since the validator in the following example will be called `CustomCartValidator`, its directory will be
+`<plugin root>/src/Core/Checkout/Cart/Custom`.
+
+Your validator has to implement the interface ´Shopware\Core\Checkout\Cart\CartValidatorInterface`.
+This forces you to also implement a `validate` method.
+
+But let's have a look at the example validator first:
+{% code title="<plugin root>/src/Core/Checkout/Cart/Custom/CustomCartValidator.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Checkout\Cart\Custom;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\CartValidatorInterface;
+use Shopware\Core\Checkout\Cart\Error\ErrorCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Swag\BasicExample\Core\Checkout\Cart\Custom\Error\CustomCartBlockedError;
+
+class CustomCartValidator implements CartValidatorInterface
+{
+    public function validate(Cart $cart, ErrorCollection $errorCollection, SalesChannelContext $salesChannelContext): void
+    {
+        foreach ($cart->getLineItems()->getFlat() as $lineItem) {
+            if (!array_key_exists('customPayLoad', $lineItem->getPayload()) || $lineItem->getPayload()['customPayload'] !== 'example') {
+                $errorCollection->add(new CustomCartBlockedError($lineItem->getId()));
+
+                return;
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+As already said, a cart validator has to implement the `CartValidatorInterface` and therefore implement a `validate` method.
+This method has access to some important parts of the checkout, such as the cart and the current sales channel context.
+Also you have access to the error collection, which may or may not contain errors from other earlier validators.
+
+In this example we're dealing with the line items and are validating them, so we're iterating over each line item.
+This example assumes that your line items got a custom payload, called `customPayLoad`, and it expects a value in there.
+
+If the condition doesn't match and the line item seems to be invalid, you'll have to add a new error to the error collection.
+You can't just use any exception here, but a class which has to extend from `Shopware\Core\Checkout\Cart\Error\Error`.
+Most likely you want to create your own error class here, which will be done in the next step.
+
+Important to note is the `return` statement afterwards.
+If you wouldn't return here, it would an error to error collection for each invalid line item, resulting in several errors displayed on the checkout or the cart page.
```

Comment:
```suggestion
If you wouldn't return here, it would add an error to error collection for each invalid line item, resulting in several errors displayed on the checkout or the cart page.
```